python/paddle/distributed/launch.py

```diff
@@ -63,6 +65,32 @@ def _print_arguments(args):
     print("------------------------------------------------")
 
 
+def find_free_ports(num):
+    def __free_port():
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+            s.bind(('', 0))
+            return s.getsockname()[1]
+
+    port_set = set()
+    step = 0
+    while True:
+        port = __free_port()
```

Comment:
if this iter get port1, will next iter get port1 again?

Reply:
No, the probability is small and the port will be filtered by the port_set.